source/channel_bootstrap.c

```diff
@@ -451,7 +451,9 @@ static void s_on_client_connection_established(struct aws_socket *socket, int er
     }
 
     if (error_code || connection_args->connection_chosen) {
-        if (connection_args->outgoing_options.domain != AWS_SOCKET_LOCAL && error_code) {
+        if ((connection_args->outgoing_options.domain == AWS_SOCKET_IPV4 ||
```

Comment:
Can we pull the domain check out into a helper function?

Reply:
I've added an s_aws_socket_domain_uses_dns.